src/rageshake/rageshake.ts

```diff
@@ -415,15 +416,19 @@ class IndexedDBLogStore {
  * @return {Promise<T[]>} Resolves to an array of whatever you returned from
  * resultMapper.
  */
-function selectQuery(store, keyRange, resultMapper) {
+function selectQuery<T>(
+    store: IDBIndex, keyRange: IDBKeyRange, resultMapper: (cursor: IDBCursorWithValue) => T,
+): Promise<T[]> {
     const query = store.openCursor(keyRange);
     return new Promise((resolve, reject) => {
         const results = [];
         query.onerror = (event) => {
+            // @ts-ignore
             reject(new Error("Query failed: " + event.target.errorCode));
         };
         // collect results
         query.onsuccess = (event) => {
+            // @ts-ignore
```

Comment:
How come we have to `@ts-ignore` here?
Wondering whether setting a type on the `event` parameter would mitigate the issue?

Reply:
I wanted to do that but this doesn't appear to be typed correctly :( Without this, it screams about `EventTarget` not having the properties it has

Reply:
![Screenshot_20210927_095628](https://user-images.githubusercontent.com/25768714/134867486-d018d59c-d84e-43a5-8ab4-d02c1627bb94.png)

For example ^

Reply:
The `event` only seems to have the type `Event`

Reply:
Right, yeah after reading the IndexedDB spec, it doesn't seem that the `onsuccess` event has a proper interface assigned to it.
People seem to be using `any` to remove the warning, but `@ts-ignore` seems quite appropriate too!